js/ui/scheduler/tooltip_strategies/tooltipStrategyBase.js

```diff
@@ -140,10 +137,12 @@ export class TooltipStrategyBase {
 
     _onListItemClick(e) {
         this.hide();
-        if(this._extraOptions.clickEvent) {
-            this._extraOptions.clickEvent(e);
-        }
-        this._options.showAppointmentPopup(e.itemData.data, false, e.itemData.currentData);
+        this._extraOptions.clickEvent && this._extraOptions.clickEvent(e);
+        this._options.showAppointmentPopup(e.itemData.data, false, this._getCurrentAppointmentData(e.itemData));
+    }
+
+    _getCurrentAppointmentData(item) {
+        return (item.settings && item.settings.targetedAppointmentData) || item.currentData || item.data;
```

Comment:
Maybe `return (item.settings?.targetedAppointmentData) || item.currentData || item.data;`?

Reply:
Why not. I will change next time